velox/common/fuzzer/ConstrainedGenerators.h

```diff
@@ -480,6 +480,23 @@ class CastVarcharInputGenerator : public AbstractInputGenerator {
   std::string generateValidPrimitiveAsString();
 };
 
+class URLInputGenerator : public AbstractInputGenerator {
+ public:
+  URLInputGenerator(size_t seed, const TypePtr& type, double nullRatio);
+
+  ~URLInputGenerator() override;
+
+  variant generate() override;
+
+ private:
+  const std::string kProtocol = "https";
+  const std::string kDomain = "www.facebook.com";
+  const std::string kPort = "8080";
+  const std::string kPath = "/test";
+  const std::string kQuery = "a=1&b=2";
+  const std::string kFragment = "test";
```

Comment:
I wonder whether we should randomly generate these strings. E.g., could you check whether valid vs. invalid protocol string affects the behavior of the url_xxx functions in Presto/Velox? Same for the other fields.

Reply:
Perhaps outside the scope of this diff, but something that would be really useful for generating URLs, JSON, JSON paths, regexes, etc. would be a tool that could take a grammar and generate random strings according to that grammar.

Short of that though, randomly generating strings (perhaps from fixed character sets) would offer better coverage.